tests/unit/SequentialQueueTest.ts

```diff
@@ -0,0 +1,148 @@
+import Onyx from 'react-native-onyx';
+import * as PersistedRequests from '@userActions/PersistedRequests';
+import ONYXKEYS from '@src/ONYXKEYS';
+import * as SequentialQueue from '../../src/libs/Network/SequentialQueue';
+import type Request from '../../src/types/onyx/Request';
+import type {ConflictActionData} from '../../src/types/onyx/Request';
+import * as TestHelper from '../utils/TestHelper';
+import waitForBatchedUpdates from '../utils/waitForBatchedUpdates';
+
+const request: Request = {
+    command: 'ReconnectApp',
+    successData: [{key: 'userMetadata', onyxMethod: 'set', value: {accountID: 1234}}],
+    failureData: [{key: 'userMetadata', onyxMethod: 'set', value: {}}],
+};
+
+describe('SequentialQueue', () => {
+    beforeAll(() => {
+        Onyx.init({
+            keys: ONYXKEYS,
+        });
+    });
+    beforeEach(() => {
+        global.fetch = TestHelper.getGlobalFetchMock();
+        return Onyx.clear().then(waitForBatchedUpdates);
+    });
+
+    it('should push one request and persist one', () => {
+        SequentialQueue.push(request);
+        expect(PersistedRequests.getLength()).toBe(1);
+    });
+
+    it('should push two requests and persist two', () => {
+        SequentialQueue.push(request);
+        SequentialQueue.push(request);
+        expect(PersistedRequests.getLength()).toBe(2);
+    });
+
+    it('should push two requests with conflict resolution and replace', () => {
```

Comment:
I think we should also add a test that queues other unrelated requests and make sure that the replaced request is in the same index as the conflicting `ReconnectApp` request.

i.e: [`OpenReport`, `AddComment`, `ReconnectApp`, `OpenReport`]

becomes [`OpenReport`, `AddComment`, `ReconnectApp`, `OpenReport`], with `ReconnectApp` still at index 2 for example